src/Mod/TechDraw/App/DrawTemplate.cpp

```diff
@@ -88,41 +80,30 @@ double DrawTemplate::getHeight() const
     return Height.getValue();
 }
 
-short DrawTemplate::mustExecute() const
-{
-    return App::DocumentObject::mustExecute();
-}
-
-/// get called by the container when a Property was changed
-void DrawTemplate::onChanged(const App::Property* prop)
-{
-    App::DocumentObject::onChanged(prop);
-}
-
 App::DocumentObjectExecReturn *DrawTemplate::execute()
 {
-    DrawPage *page = nullptr;
-    std::vector<App::DocumentObject*> parent = getInList();
-    for (std::vector<App::DocumentObject*>::iterator it = parent.begin(); it != parent.end(); ++it) {
-        if ((*it)->getTypeId().isDerivedFrom(DrawPage::getClassTypeId())) {
-            page = dynamic_cast<TechDraw::DrawPage *>(*it);
-        }
-    }
-
-    if(page) {
-        page->Template.touch();     //if you are on a page, execute yourself???
-    }
+//    DrawPage *page = nullptr;
+//    std::vector<App::DocumentObject*> parent = getInList();
+//    for (std::vector<App::DocumentObject*>::iterator it = parent.begin(); it != parent.end(); ++it) {
+//        if ((*it)->getTypeId().isDerivedFrom(DrawPage::getClassTypeId())) {
+//            page = dynamic_cast<TechDraw::DrawPage *>(*it);
+//        }
+//    }
+
+//    if(page) {
+//        page->Template.touch();
+//    }
 
     return App::DocumentObject::execute();
 }
 
 DrawPage* DrawTemplate::getParentPage() const
 {
-    TechDraw::DrawPage* page = nullptr;
-    std::vector<App::DocumentObject*> parent = getInList();
-    for (std::vector<App::DocumentObject*>::iterator it = parent.begin(); it != parent.end(); ++it) {
-        if ((*it)->getTypeId().isDerivedFrom(DrawPage::getClassTypeId())) {
-            page = static_cast<TechDraw::DrawPage *>(*it);
+    TechDraw::DrawPage* page(nullptr);
+    std::vector<App::DocumentObject*> parents = getInList();
+    for (auto& obj : parents) {

```

Comment:
If you're trying to get the last page in parents, it might be clearer to use a reverse iterator and return right after you find it. Otherwise it reads like this might be an error, and you meant to find the first, not the last.

Reply:
I just changed the iterator loop with range based loop.  Never noticed the missing break.  Again good eye.  Thanks.